Multiple unit tests failing with "cannot guess the engine" since xarray 0.18.0 release

**Describe the bug**
Since last week, twelve unit tests are failing in CI and in newly installed local environments:

```
FAILED test/core/test_timeslice.py::TimeSliceTest::test_find_time_slice - ValueError: cannot guess the engine, try passing one explicitly
FAILED test/core/gen/test_gen.py::DefaultProcessTest::test_handle_360_lon - ValueError: cannot guess the engine, try passing one explicitly
FAILED test/core/gen/test_gen.py::DefaultProcessTest::test_input_txt - ValueError: cannot guess the engine, try passing one explicitly
FAILED test/core/gen/test_gen.py::DefaultProcessTest::test_insert_packed_zarr - ValueError: cannot guess the engine, try passing one explicitly
FAILED test/core/gen/test_gen.py::DefaultProcessTest::test_process_chunked_zarr - ValueError: cannot guess the engine, try passing one explicitly
FAILED test/core/gen/test_gen.py::DefaultProcessTest::test_process_compressed_zarr - ValueError: cannot guess the engine, try passing one explicitly
FAILED test/core/gen/test_gen.py::DefaultProcessTest::test_process_inputs_append_multiple_zarr - ValueError: cannot guess the engine, try passing one explicitly
FAILED test/core/gen/test_gen.py::DefaultProcessTest::test_process_inputs_insert_multiple_zarr - ValueError: cannot guess the engine, try passing one explicitly
FAILED test/core/gen/test_gen.py::DefaultProcessTest::test_process_inputs_replace_multiple_zarr - ValueError: cannot guess the engine, try passing one explicitly
FAILED test/core/gen/test_gen.py::DefaultProcessTest::test_process_inputs_single_zarr - ValueError: cannot guess the engine, try passing one explicitly
FAILED test/core/gen/test_gen.py::DefaultProcessTest::test_process_packed_zarr - ValueError: cannot guess the engine, try passing one explicitly
FAILED test/core/gen/test_gen.py::DefaultProcessTest::test_replace_packed_zarr - ValueError: cannot guess the engine, try passing one explicitly
```

This seems to be connected with the release of xarray 0.18.0 last week, since the tests are all still passing with xarray 0.17.0

**To Reproduce**
Steps to reproduce the behavior:
1. Clone the current xcube repository.
2. Create a new conda environment using xcube's `environment.yml` and activate it.
3. Run the test suite with `pytest`.

**Expected behavior**
All tests pass.

**Desktop (please complete the following information):**
 - OS: Ubuntu 20.04, Mac

--- a/xcube/core/timeslice.py
+++ b/xcube/core/timeslice.py
@@ -51,7 +51,10 @@ def find_time_slice(store: Union[str, MutableMapping],
         # ValueError raised if cube store does not exist
         try:
             cube = xr.open_dataset(store)
-        except FileNotFoundError:
+        except (FileNotFoundError, ValueError):
+            # If the zarr directory does not exist, open_dataset raises a
+            # FileNotFoundError (with xarray <= 0.17.0) or a ValueError
+            # (with xarray 0.18.0).
             return -1, 'create'
 
     # TODO (forman): optimise following naive search by bi-sectioning or so

